extensions/icd/actions/tfu/icd.ts

```diff
@@ -0,0 +1,176 @@
+import { type Setting, type Action } from '@awell-health/extensions-core'
+import { Category } from '@awell-health/extensions-core'
+import axios from 'axios'
+import { isEmpty, isNil } from 'lodash'
+import z from 'zod'
+import { FieldSchema, fields } from './config/fields'
+import { dataPoints } from './config/dataPoints'
+
+const settings = {} satisfies Record<string, Setting>
+const SettingSchema = z.unknown()
+
+const PayloadSchema = z.object({
+  fields: FieldSchema,
+  settings: SettingSchema,
+})
+
+export const icd: Action<
+  typeof fields,
+  typeof settings,
+  keyof typeof dataPoints
+> = {
+  key: 'icd',
+  category: Category.DEMO,
+  title: 'ICD-10 Code Lookup',
+  description: 'Lookup ICD-10 codes and return the parent disease',
+  fields,
+  previewable: true,
+  dataPoints,
+  onActivityCreated: async (payload, onComplete): Promise<void> => {
+    const {
+      fields: { icd_codes },
+    } = PayloadSchema.parse(payload)
+    // if no codes, just return blank
+    if (icd_codes.length === 0) {
+      await onComplete({
+        data_points: {
+          codes: JSON.stringify([]),
+          stringResponse: 'Blank',
+        },
+      })
+      return
+    }
+
+    const accessToken = await grabAccessToken()
+
+    const icdCodes = await grabICDCodes(accessToken, icd_codes)
+
+    const resp = icdCodes.some(checkICDCode)
+
+    await onComplete({
+      data_points: {
+        codes: JSON.stringify(icdCodes),
+        stringResponse: resp ? 'TFU' : 'TCM',
+      },
+    })
+  },
+}
+
+const ICDSchema = z
+  .object({
+    parent: z.array(z.string()),
+    title: z.object({
+      '@language': z.string(),
+      '@value': z.string(),
+    }),
+  })
+  .refine(({ parent }) => parent.length !== 0)
+  .transform((data) => {
+    return {
+      parent: data.parent[0],
+      title: data.title['@value'],
+    }
+  })
+  .refine(({ parent }) => !isNil(parent) && !isEmpty(parent))
+  .refine(({ title }) => !isNil(title) && !isEmpty(title))
+
+interface LanguageResponse {
+  '@language': string
+  '@value': string
+}
+
+interface QueryResponse {
+  '@context': string
+  '@id': string
+  parent: string[]
+  child: string[]
+  browserUrl: string
+  code: string
+  title: LanguageResponse
+  inclusion: Array<{ label: LanguageResponse }>
+  exclusion: Array<{ label: LanguageResponse }>
+  classKind: string
+}
+
+const checkICDCode = (code: string): boolean => {
```

Comment:
Is this the only way of doing this? Cant we actually retrieve the codes like this?

<img width="130" alt="image" src="https://github.com/awell-health/awell-extensions/assets/3439020/19f4d76a-a373-454d-bc4c-b91a57f841f0">